web/src/views/DandisetLandingView/DandisetLandingView.vue

```diff
@@ -301,6 +312,10 @@ export default {
         }
       },
     },
+    // TODO move this somewhere more sensical
+    async meta(val) {
+      this.$store.dispatch('publish/fetchPublishedVersions', val.identifier);
```

Comment:
How about putting this at the end of the `id` watcher, which initializes this component on creation or navigation?